draw_flip_book/lib/main.dart

```diff
@@ -79,31 +79,83 @@ class _FlipBookPageState extends State<FlipBookPage> {
     // fast, so that you can just rebuild anything that needs updating rather
     // than having to individually change instances of widgets.
     return new Scaffold(
-      body: buildGestureDetector(
+      body: _buildGestureDetector(
         context,
-        Stack(
-          alignment: Alignment.center,
-          children: <Widget>[
-            // TODO: Generalize/Scale
-            _buildPositionedFrame(context, key1, points1, true, Colors.green),
-            _buildPositionedFrame(
-                context, key2, points2, _isVisible2, Colors.lightBlue),
-            _buildPositionedFrame(
-                context, key3, points3, _isVisible3, Colors.amberAccent),
-          ],
+        Container(
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            children: <Widget>[
+              Container(
+                height: 500,
+                child: _framesStack(context),
+              ),
+              Expanded(
+                child: Container(child: _buttonRow()),
+              )
+            ],
+          ),
         ),
       ),
-      floatingActionButton: FloatingActionButton(
-          onPressed: () {
-            setState(() {
-              _toggleFramesVisibility();
-            });
-          },
-          child: Icon(Icons.skip_next)),
     );
   }
 
-  GestureDetector buildGestureDetector(BuildContext context, Widget child) {
+  Row _buttonRow() {
+    return Row(
+      mainAxisSize: MainAxisSize.max,
+      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+      children: <Widget>[
+        Container(
+          child: FloatingActionButton(
+            onPressed: () {
+              setState(() {
+                _toggleFramesVisibility();
+              });
+            },
+            child: Icon(Icons.navigate_next),
+          ),
+        ),
+        Container(
+          child: FloatingActionButton(
+            onPressed: () {
+              setState(() {
+                // TODO: loop all the frames in sequence
+              });
+            },
+            child: Icon(Icons.play_arrow),
+          ),
+        ),
+        Container(
+          child: FloatingActionButton(
+            onPressed: () {
+              setState(() {
+                _clear();
+                currentFrame = 3;
+                _isVisible3 = true;
+                _replayFrames = false;
+              });
+            },
+            child: Icon(Icons.clear),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Stack _framesStack(BuildContext context) {
+    return Stack(
```

Comment:
For simple members, consider using => (not picky though).
https://dart.dev/guides/language/effective-dart/usage#consider-using--for-simple-members

Reply:
nice, 8db9d38